File: back-end/src/main/java/admin/parkWise/administration/models/ParkingLot.java
package admin.parkWise.administration.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "parking_lots")
public class ParkingLot {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "name", nullable = false)
    @JsonProperty("name")
    private String name;

    @Column(name = "address", nullable = false)
    @JsonProperty("address")
    private String address;
    @Column(name = "lat", nullable = false)
    @JsonProperty("lat")
    private Float lat;
    @Column(name = "lng", nullable = false)
    @JsonProperty("lng")
    private Float lng;

    @Column(name = "total_parking_spaces")
    @JsonProperty("totalParkingSpaces")
    private Integer totalParkingSpaces;

    @Column(name = "available_parking_spaces")
    @JsonProperty("availableParkingSpaces")
    private Integer availableParkingSpaces;
}

File: front_end/lib/model/ParkingLot.dart
// lib/models/parking_lot.dart

class ParkingLot {
  final int id; 
  final String name;
  final String address;
  final double lat;
  final double lng;
  final int? totalParkingSpaces; 
  final int? availableParkingSpaces; 

  const ParkingLot({
    required this.id,
    required this.name,
    required this.address,
    required this.lat,
    required this.lng,
    required this.totalParkingSpaces,
    required this.availableParkingSpaces,
  });

  factory ParkingLot.fromJson(Map<String, dynamic> json) {
    return ParkingLot(
      id: json['id'] as int,
      name: json['name'] as String,
      address: json['address'] as String,
      lat: (json['lat'] as num).toDouble(),
      lng: (json['lng'] as num).toDouble(),
      totalParkingSpaces: json['totalParkingSpaces'] as int?,
      availableParkingSpaces: json['availableParkingSpaces'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'address': address,
      'lat': lat,
      'lng': lng,
      'totalParkingSpaces': totalParkingSpaces,
      'availableParkingSpaces': availableParkingSpaces,
    };
  }

  @override
  String toString() {
    return 'ParkingLot(id: $id, name: "$name", address: "$address", '
        'lat: $lat, lng: $lng, totalParkingSpaces: $totalParkingSpaces, '
        'availableParkingSpaces: $availableParkingSpaces)';
  }
}
